cypress/e2e/GUI/customer_details.cy.js

```diff
@@ -0,0 +1,50 @@
+/// <reference types="Cypress" />
+describe('Detalhes do cliente', () => {
+    beforeEach(() => {
+        const dateInThePast = new Date(Date.UTC(1983, 6, 7))
+            cy.clock(dateInThePast)
+            cy.visit('/') 
+    });
+    context('Quando valido a saudação por default e a data', () => {
+        it('Então deve conter "Hi there" e a data "Wed Jul 06 1983"', () => {
+            cy.wait('@pag1limit10all')
+            cy.get('[data-testid="table"] tbody tr').then(($confirmElement) => {
+                if($confirmElement.length > 0) {
+                    cy.get('[data-testid="table"] > :nth-child(1)')
+                    .should('exist')
+                    .and('contain.text', 'Hi there');
+
+                    cy.get('main > .table-container > p > b:nth-child(2)')
+                    .should('exist')
+                    .and('contain.text', 'Wed Jul 06 1983')
+                    .and('be.visible');
+                }
+            });
+        });
+    });
+    context('Quando valido a saudação personalisada e a data ', () => {
```

Comment:
Se for pra usar DADO/QUANDO/ENTÃO, use assim:

* DADO - utilizado para descrever a pré-condição do teste (ex.: `Dado que visito a página inicial da aplicação EngageSphere`)
* QUANDO - utilizado para descrever a ação (ou ações) do teste que exercitam certo comportamento (ex.: `Quando filtro por usuários com tamanho Médio`)
* ENTÃO - utilizado para descrever o resultado esperado (ex.: `Então só usuários com tamanho Médio são exibidos na tabela de clientes`)